UnicornRender/video/include/unicorn/video/Primitives.hpp

```diff
@@ -10,41 +10,54 @@
 #include <unicorn/utility/SharedMacros.hpp>
 #include <unicorn/video/Mesh.hpp>
 
+#include <string>
+#include <list>
+
 namespace unicorn
 {
 namespace video
 {
-/** @brief  Fills meshes with geometric primitive's data */
+/** @brief  Fills meshes with geometric primitives data */
 class Primitives
 {
 public:
     /**
     *  @brief Fills given mesh with cube data
     *
-    *  @param[out] mesh    geometry mesh
-    *  @return reference to updated mesh
+    *  @param[in, out] mesh geometry mesh
+    *  @return pointer to filled mesh
     */
-    UNICORN_EXPORT static Mesh& Box(Mesh& mesh);
+    UNICORN_EXPORT static Mesh* Box(Mesh* mesh = nullptr);
 
     /**
     *  @brief Fills given mesh with quad data
     *
-    *  @param[out] mesh    geometry mesh
-    *  @return reference to updated mesh
+    *  @param[in, out] mesh geometry mesh
+    *  @return pointer to updated mesh
     */
-    UNICORN_EXPORT static Mesh& Quad(Mesh& mesh);
+    UNICORN_EXPORT static Mesh* Quad(Mesh* mesh = nullptr);
 
     /**
     *  @brief Fills given mesh with sphere data
     *
-    *  @param[out] mesh geometry mesh
     *  @param[in] radius radius of sphere (> 0)
     *  @param[in] rings horizontal slices count (>= 4)
     *  @param[in] sectors vertical slices count (>= 4)
+    *  @param[out] mesh geometry mesh
     *  @attention `rings * sectors` must be less than `std::vector<Vertex>::max_size()`
-    *  @return reference to updated mesh
+    *  @return pointer to updated mesh
+    */
+    UNICORN_EXPORT static Mesh* Sphere(float radius, uint32_t rings, uint32_t sectors, Mesh* mesh = nullptr);
+
+    /**
+    *  @brief Loading model from file, fills materials and returns list of pointers to meshes
+    *
+    *  Supported formats: gltf 2.0 (only gltf for now); obj
```

Comment:
I don't get it, what is actually supported?
- gltf 2.0, obj
- gltf, obj
- gltf 2.0 only
- gltf